js/src/makegrid.js

```diff
@@ -320,6 +380,35 @@ var makeGrid = function (t, enableResize, enableReorder, enableVisib, enableGrid
             }
         },
 
+        /**
+         * Store the column width
+         *
+         * @param n zero-based column index
+         * @param nw new width of the column in pixel
+         */
+        storeColumnWidth: function (n, nw) {
+            var key   = g.dbTableHash + '_column_width';
+            var name = $('.pma_table').eq(0).find('th').eq(n + 1)[0].dataset.column;
+            var value;
+
+            var data = g.getStorageData(key);
+
+            if (data === null) {
+                var obj = { name:nw };
+                value = JSON.stringify(obj);
+            } else {
+                data = JSON.parse(data);
+                data[name] = nw;
+                value = JSON.stringify(data);
+            }
```

Comment:
```suggestion
            var encodeData = data === null ? {} : JSON.parse(data);
            encodeData[name] = nw;
            value = JSON.stringify(encodeData);
```

This should avoid un-needed complexity